### PR TITLE
feat: add support for query offset

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY
In order to support server side pagination, support for `row_offset` needs to be added to to complement `row_limit` in `QueryObject`.

### TEST PLAN
CI + new tests

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/charts/schemas.py [hunk 1]
```diff
@@ -664,6 +664,7 @@ class ChartDataQueryObjectSchema(Schema):
     row_limit = fields.Integer(
         description='Maximum row count. Default: `config["ROW_LIMIT"]`',
     )
+    row_offset = fields.Integer(description="Number of rows to skip. Default: `0`",)
```
Comment (dpgaspar) [inline superset/charts/schemas.py:667]: consider enforcing `row_offset` is >0
```
validate=[Range(min=0, error=_("row_offset must be greater than 0"))])
```

File: superset/charts/schemas.py [hunk 2]
```diff
@@ -664,6 +664,7 @@ class ChartDataQueryObjectSchema(Schema):
     row_limit = fields.Integer(
```
Comment (dpgaspar) [inline superset/charts/schemas.py:664]: consider enforcing `row_limit` is >0
```
validate=[Range(min=0, error=_("row_limit must be greater than 0"))])
```

File: tests/charts/api_tests.py [hunk 3]
```diff
@@ -634,19 +636,43 @@ def test_get_charts_no_data_access(self):
...
         self.assertEqual(data["result"][0]["rowcount"], 100)
 
+    def test_chart_data_limit_offset(self):
+        """
+        Query API: Test chart data query with limit and offset
+        """
+        self.login(username="admin")
+        table = self.get_table_by_name("birth_names")
+        request_payload = get_query_context(table.name, table.id, table.type)
+        request_payload["queries"][0]["row_limit"] = 5
```
Comment (dpgaspar) [inline tests/charts/api_tests.py:658]: Would be nice to a couple of simple tests with invalid `row_limit` and `row_offset`. negative values, bigger then `SAMPLES_ROW_LIMIT`

**Review by dpgaspar — COMMENTED** [review]
Looks good, just a couple of comments

File: tests/charts/api_tests.py (hunk 3 above)
Comment (villebro) [reply tests/charts/api_tests.py:661]: Added schema validation tests (`tests/chart/schema_tests.py`) for validation of `row_limit ` and `row_offset` and e2e tests (`tests/chart/api_tests.py`) to ensure regular and sample row limits produce correct row counts.

**Review by dpgaspar — APPROVED** [review]

**Pull request merged**